### PR TITLE
feat: enable rook-ceph monitoring

### PR DESCRIPTION
**Pull request by Azhovan**
**What problem does this PR solve?**:
Enabling monitoring for rook-ceph 

**Which issue(s) does this PR fix?**:
https://d2iq.atlassian.net/browse/D2IQ-93139

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**If the PR adds a version bump, does it add a breaking change in License**:

- [ ] No License Change (or NA).

### CODE REVIEW

**Review by takirala — APPROVED** [review]

**Review by gracedo — APPROVED** [review]
does the chart install grafana dashboards? or would we need to pull in some upstream dashboards?

**Comment by takirala** [comment]
The chart does not have any grafana dashboards. Some research needs to be done to see which dashboard we can use - perhaps https://grafana.com/grafana/dashboards/7056-ceph-cluster/ is a good start

**Comment by takirala** [comment]
@Azhovan lets make sure the promethues rules are working as expected and a dashboard is able to collect all relevant metrics before resolving this ticket.

**Comment by Azhovan** [comment]
> @Azhovan lets make sure the promethues rules are working as expected and a dashboard is able to collect all relevant metrics before resolving this ticket.

Screenshot_20221019_142402

**Comment by Azhovan** [comment]
@takirala  @gracedo 
Screenshot_20221019_161750

**Comment by gracedo** [comment]
@Azhovan Can you please export the json of that dashboard and add it to the `rook-ceph` app? You can take a look at how we define grafana dashboards in other apps e.g. https://github.com/mesosphere/kommander-applications/tree/main/services/traefik/10.30.1/grafana-dashboards

**Review by takirala — COMMENTED** [review]
I think `rook-ceph-cluster` is a better location for the dashboard instead of `rook-ceph` - Because without the cluster, the dashboard might not have much data (or pretty much no data at all), so lets move it into `rook-ceph-cluster/grafana-dashboards` directory.

Also, the `grafana-dashboards` dir needs to be refernced from `<app-name>/<version>/kustomization.yaml` file inorder to be rendered (e.g.: https://github.com/mesosphere/kommander-applications/blob/main/services/velero/3.3.0/kustomization.yaml#L5)

File: services/rook-ceph/1.10.3/grafana-dashboards/rook-ceph.json [hunk 1]
```diff
@@ -0,0 +1,5484 @@
...
+      "6h",
+      "12h",
+      "24h",
+      "2d",
+      "7d",
+      "30d"
+    ]
+  },
+  "timezone": "browser",
+  "title": "Ceph - Cluster",
```
Comment (takirala) [inline services/rook-ceph/1.10.3/grafana-dashboards/rook-ceph.json]: Lets keep this naming as `Rook Ceph Cluster` to keep it consistent with AppDeployment name

Comment (Azhovan) [reply services/rook-ceph/1.10.3/grafana-dashboards/rook-ceph.json]: done.

File: services/rook-ceph-cluster/1.10.3/grafana-dashboards/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: ${releaseNamespace}
+configMapGenerator:
+  - name: grafana-dashboards-rook-ceph
```
Comment (gracedo) [inline services/rook-ceph-cluster/1.10.3/grafana-dashboards/kustomization.yaml:5]:
```suggestion
  - name: grafana-dashboards-rook-ceph-cluster
```

File: services/rook-ceph-cluster/1.10.3/grafana-dashboards/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    namespace: ${releaseNamespace}
+    files:
+      - rook-ceph.json
```
Comment (gracedo) [inline services/rook-ceph-cluster/1.10.3/grafana-dashboards/kustomization.yaml:8]:
```suggestion
      - rook-ceph-cluster.json
```

can we rename the json file to `rook-ceph-cluster.json` just to be consistent?

File: services/rook-ceph-cluster/1.10.3/grafana-dashboards/rook-ceph.json [hunk 4]
```diff
@@ -0,0 +1,5484 @@
...
+      "6h",
+      "12h",
+      "24h",
+      "2d",
+      "7d",
+      "30d"
+    ]
+  },
+  "timezone": "browser",
+  "title": "Rook Ceph Cluster",
```
Comment (gracedo) [inline services/rook-ceph-cluster/1.10.3/grafana-dashboards/rook-ceph.json:5480]:
```suggestion
  "title": "Platform Apps / Rook Ceph Cluster",
```

(we prefix our platform apps dashboard names just for some organization)

File: services/rook-ceph-cluster/1.10.3/grafana-dashboards/kustomization.yaml (hunk 3 above)
Comment (Azhovan) [reply services/rook-ceph-cluster/1.10.3/grafana-dashboards/kustomization.yaml]: thanks

File: services/rook-ceph-cluster/1.10.3/grafana-dashboards/kustomization.yaml (hunk 2 above)
Comment (Azhovan) [reply services/rook-ceph-cluster/1.10.3/grafana-dashboards/kustomization.yaml]: thanks

File: services/rook-ceph-cluster/1.10.3/grafana-dashboards/rook-ceph.json (hunk 4 above)
Comment (Azhovan) [reply services/rook-ceph-cluster/1.10.3/grafana-dashboards/rook-ceph.json]: thanks

**Pull request merged**